### PR TITLE
Vagrant: Allow env override of IP addresses

### PR DESCRIPTION
**Pull request by marun**
MASTER_IP and MINION_IP_BASE are hard-coded in vagrant's
config-default.sh, and the values correspond with the subnet used by
virtualbox.  On hosts that have both virtualbox and another provider
installed (e.g. vmware), attempting to deploy kubernetes with the
non-virtualbox provider is likely to result in VM's on the wrong
network.  This change allows the addresses to be overridden via
the environment.

### CODE REVIEW

**Comment by zmerlynn** [comment]
@k8s-bot: ok to test

**Comment by zmerlynn** [comment]
@marun: Thanks for the submission. In order to proceed, I'll need to sort out the same CLA issue as https://github.com/GoogleCloudPlatform/kubernetes/pull/10601#issuecomment-117434067. Thanks!

File: cluster/vagrant/config-default.sh [hunk 1]
```diff
@@ -21,14 +21,14 @@ NUM_MINIONS=${NUM_MINIONS-"1"}
 export NUM_MINIONS
 
 # The IP of the master
-export MASTER_IP="10.245.1.2"
-export KUBE_MASTER_IP="10.245.1.2"
+export MASTER_IP=${MASTER_IP-"10.245.1.2"}
+export KUBE_MASTER_IP=$MASTER_IP
```
Comment (zmerlynn) [inline cluster/vagrant/config-default.sh]: It's pedantic, but `s/$MASTER_IP/${MASTER_IP}/`. The dominant bash style I've been trying to enforce, though it's waivering, is for `${TOKEN}`, regardless of context.

**Comment by zmerlynn** [comment]
LGTM except a nit.

**Comment by zmerlynn** [comment]
LGTM

**Comment by derekwaynecarr** [comment]
LGTM

**Comment by bprashanth** [comment]
@marun please provide a risk/value assessment if you need this for 1.0 (and a bug, if there is one); if not we'll let the pr bake for a week till we're done with the release.

**Comment by marun** [comment]
bprashanth: The risk here is low, since behavior is not impacted without explicit configuration of the environment variables.  I'm not sure there's much point in trying to get it into 1.0, though, since I'm assuming that vagrant deployment is primarily a developer convenience and developers are likely to pull from master anyway.

**Comment by yujuhong** [comment]
> bprashanth: The risk here is low, since behavior is not impacted without explicit configuration of the environment variables. I'm not sure there's much point in trying to get it into 1.0, though, since I'm assuming that vagrant deployment is primarily a developer convenience and developers are likely to pull from master anyway.

Given the reply, I'm moving this to post-1.0 for now :)

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Pull request merged**